zebra-consensus/src/primitives/groth16/params.rs

```diff
@@ -67,31 +73,56 @@ impl Groth16Parameters {
         //       (we check them on load anyway)
         if !sapling_spend_path.exists() || !sapling_output_path.exists() {
             tracing::info!("downloading Zcash Sapling parameters");
-            let new_sapling_paths =
-                zcash_proofs::download_sapling_parameters(Some(PARAMETER_DOWNLOAD_TIMEOUT))
-                    .unwrap_or_else(|error| {
-                        panic!(
-                            "error downloading Sapling parameter files: {:?}. {}",
-                            error,
-                            Groth16Parameters::failure_hint()
-                        )
-                    });
-            assert_eq!(sapling_spend_path, new_sapling_paths.spend);
-            assert_eq!(sapling_output_path, new_sapling_paths.output);
+
+            let mut sapling_retries = 0;
+            while sapling_retries < PARAMETER_DOWNLOAD_MAX_RETRIES {
+                let new_sapling_paths =
+                    zcash_proofs::download_sapling_parameters(Some(PARAMETER_DOWNLOAD_TIMEOUT));
+
+                if new_sapling_paths.is_err() {
+                    sapling_retries += 1;
+                    tracing::info!("error downloading Zcash Sapling parameters, retrying");
```

Comment:
I tried to put the underlying error in the message but i had compiler issues as:

```
 doesn't satisfy `SaplingParameterPaths: std::fmt::Debug`
 ```

where deriving `Debug` is not an option as it is outside or code. I think there should be a way to do this in our side but i was not able to find how.

Reply:
I actually wrote that code, and my `zcash_proofs` PR is still waiting for review, so it was easy to fix.

You can get those changes in Zebra once the dependency update in #3308 merges.